### PR TITLE
Update workflow to include elixir 1.10 to 1.13 and OTP 22 to 24

### PR DESCRIPTION
**Pull request by warmwaffles**
This also separates the linting from the testing phase of the workflows to ensure that we only care about the latest linter rules and try to ensure as much backwards compatibility as possible.

### CODE REVIEW

**Comment by warmwaffles** [comment]
Here's the example of how the workflow runs https://github.com/warmwaffles/file_store/actions/runs/1743495054

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,54 +1,80 @@
...
-      - name: Setup Elixir
-        uses: actions/setup-elixir@v1
+      - uses: erlef/setup-beam@v1
         with:
-          otp-version: "22.3.4"
+          otp-version: ${{ matrix.otp }}
           elixir-version: ${{ matrix.elixir }}
-
+      - uses: egor-tensin/vs-shell@v2
+        if: runner.os == 'Windows'
```
Comment (rzane) [inline .github/workflows/build.yml]: Since we're not testing against Windows, would you mind removing this bit?

**Review by rzane — COMMENTED** [review]
Looks great other than that one thing!

Did you, by chance, attempt to run the tests against Windows? It'd be nice if it just worked, but I haven't made any deliberate attempt to support Windows as of yet.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,54 +1,80 @@
...
+          elixir-version: "1.13"
+      - uses: actions/cache@v2
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: ${{ runner.os }}-mix-
+      - run: mix deps.get
+      - run: mix deps.compile
+      - run: mix format --check-formatted
+      - run: mix deps.unlock --check-unused
```
Comment (rzane) [inline .github/workflows/build.yml:25]: Oooh, I like this one 👍

File: .github/workflows/build.yml (hunk 1 above)
Comment (warmwaffles) [reply .github/workflows/build.yml]: Yea

Comment (warmwaffles) [reply .github/workflows/build.yml]: I can try spinning up windows real quick

Comment (warmwaffles) [reply .github/workflows/build.yml]: No dice on running windows. 

> Container operations are only supported on Linux runners

If this was just a pure compile and simple unit test I think we could get it to work. But alas this needs those integration tests.

Comment (warmwaffles) [reply .github/workflows/build.yml]: Removed this portion of the workflow.

Comment (rzane) [reply .github/workflows/build.yml]: Okay, no problem, we can always add that later 👍

**Pull request merged**